framework/src/bundle/BundleResource.cpp

```diff
@@ -137,20 +129,9 @@ void BundleResource::InitializeChildren()
   }
 }
 
-BundleResource::~BundleResource()
-{
-  if (!--d->ref)
-    delete d;
-}
-
 BundleResource& BundleResource::operator=(const BundleResource& resource)
 {
-  BundleResourcePrivate* curr_d = d;
   d = resource.d;
-  ++d->ref;
-
-  if (!--curr_d->ref)
-    delete curr_d;
 
   return *this;
 }
```

Comment:
can the copy assignment operator be `=default`? I don't think we need a user-defined implementation for this now.